src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -447,47 +487,64 @@ public Response get(String json) throws Exception
 		catch (JsonSyntaxException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addError(e.getMessage()).build();
 		}
 		catch (QueryException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addError(e.getMessage()).build();
 		}
 		catch (BeanValidationException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addErrors(e.getErrorMessages()).build();
 		}
 		catch (MaxRowKeysForQueryExceededException e) {
 			logger.error("Query failed with too many rows", e);
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addError(e.getMessage()).build();
 		}
 		catch (MemoryMonitorException e)
 		{
 			logger.error("Query failed.", e);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			Thread.sleep(1000);
 			System.gc();
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
 		catch (IOException e)
 		{
 			logger.error("Failed to open temp folder "+datastore.getCacheDir(), e);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
 		catch (Exception e)
 		{
 			logger.error("Query failed.", e);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
 		catch (OutOfMemoryError e)
 		{
 			logger.error("Out of memory error.", e);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
 		finally
 		{
+			scope.close();
```

Comment:
I feel using try-with-resources would be a safer/less error prone approach here, also will be a consistent implementation.

Reply:
I agree.

@vibhory2j It may be a good idea to follow the recommended pattern for scope activation and management here: https://github.com/opentracing/opentracing-java#starting-a-new-span

Scopes can close without finishing the span, leaving it available in the catch/finally blocks.  The scope is auto-closed, but no longer necessary as that only removes it from the ScopeManager for propagation purposes.

Refactoring the activateScope(...) method to something that just returns the configured and optionally already started Span.